patches/minecraft/net/minecraft/util/registry/WorldSettingsImport.java.patch

```diff
@@ -0,0 +1,10 @@
+--- a/net/minecraft/util/registry/WorldSettingsImport.java
++++ b/net/minecraft/util/registry/WorldSettingsImport.java
+@@ -231,6 +231,7 @@
+ 
+          public <E> DataResult<Pair<E, OptionalInt>> func_241879_a(DynamicOps<JsonElement> p_241879_1_, RegistryKey<? extends Registry<E>> p_241879_2_, RegistryKey<E> p_241879_3_, Decoder<E> p_241879_4_) {
+             JsonElement jsonelement = this.field_244349_a.get(p_241879_3_);
++            jsonelement.getAsJsonObject().addProperty("forge:resourcelocation", p_241879_3_.func_240901_a_().toString());
```

Comment:
You need a NULL check here.
If the file does not exist or is malformed this jsonElement is null.